First pass JS cleanup - through opus.js

rfrenchseti opened the pull request:
Note this branch is based off Debby's c035_cart branch to reduce future merge problems.

- Moved all templates in apps/ui to apps/ui/templates/ui to be consistent with other apps. No templates were actually changed. Unused templates were deleted.
- Major cleanup of opus.js and associated changes to other JS files.
  - Reorg to make like functions together
  - Style changes to variable names and functions (camelCase, proper verbs in function names, etc.)
  - First pass at following JSDoc function document style (/** ... */)
  - Fixed bug in search tab spinners when tabs switched quickly
  - Removed dead deferred object code
  - Cleaned up retrieval of menu when reset buttons pressed

rfrenchseti committed:
Update ring_geo definitions

rfrenchseti committed:
Merge remote-tracking branch 'upstream/new_ui' into RF_definitions_190510

rfrenchseti committed:
Fix obs_ring_geo tests

rfrenchseti committed:
Update definitions for Cassini and ISS.

rfrenchseti committed:
Clean up Cassini mission definitions.

rfrenchseti committed:
Update CIRS and UVIS definitions.

rfrenchseti committed:
Update VIMS tooltips

juzen2003 committed:
Commit before switching branches

juzen2003 committed:
Merge remote-tracking branch 'upstream/new_ui' into new_ui_infiniteScroll_issues

juzen2003 committed:
Update code in renderGalleryAndTable, onUpdateSlider, and event handler function for ps-scroll-up

juzen2003 committed:
Add galleryAndTabPSLength to fix the ps length in gallery & table

rfrenchseti committed:
Move ui templates to ui/templates. Partial cleanup of opus.js, browse.js, widgets.js.

astrocfi committed:
Merge branch 'c035_cart' of https://github.com://dstopp/pds-opus into RF_JS_cleanup_190515

juzen2003 committed:
Update checkScroll to trigger ps-scroll-up when it's above scroll up threshold point, this will alow us to hold down page up key and keep data loading

rfrenchseti committed:
More JS cleanup

rfrenchseti committed:
More JS cleanup

rfrenchseti committed:
More JS cleanup

juzen2003 committed:
Clean up codes, commit before updating updateSliderHandle

rfrenchseti committed:
Update Galileo and GOSSI definitions.

rfrenchseti committed:
Finish cleanup of opus.js and all the other JS files it touched.

rfrenchseti committed:
Update VIMS with Mark comments

dstopp commented on opus/application/static_media/js/opus.js:
```diff
@@ -114,227 +141,284 @@ var opus = {
         o_browse.hideMenu();
 
-        // close any open modals
+        // Close any open modals
```
what about the metadata modal? do we need to worry about closing that here as well?  if so, do we need a "closeModals' function?

rfrenchseti replied:
There's a separate handler in browse.js for ESC on the metadata modal and also the select metadata modal. It also does things like stop the range select and, for no obvious reason, close the help pane. I think the right thing to do is to leave the metadata modal and select metadata modal in browse.js because they have other actions associated with them (like asking for confirmation closing the select metadata...eventually), and move the close help pane to opus.js. It's sort of a tricky decision because the confirm modals are scattered all over the place, so it's difficult to have a handler for each namespace separately. Putting them in opus.js seems like the best choice.

juzen2003 committed:
updateSliderHandle with math for both gallery & table view to obtain startObs

juzen2003 committed:
Commit before replacing loop with math in updateSliderHandle

juzen2003 committed:
Update updateSliderHandle, replace loop with math

juzen2003 committed:
Update the math in updateSliderHandle, now after resizing the window, when switching between table and gallery, slider value will not change

rfrenchseti committed:
Merge branch 'dstopp-c035_cart' into RF_JS_cleanup_190515

rfrenchseti committed:
Merge branch 'c035_cart' of https://github.com/dstopp/pds-opus into RF_JS_cleanup_190515

rfrenchseti committed:
Move closing of help pane to opus.js

juzen2003 committed:
Clean up codes

juzen2003 committed:
Rename galleryAndTabPSLength to galleryAndTablePSLength

dstopp committed:
minor cleanup

astrocfi committed:
git push origin new_uiMerge branch 'dstopp-c035_cart' into new_ui

juzen2003 committed:
Merge with upstream/new_ui

rfrenchseti committed:
Update product types, COISS,. GOSSI, and Cassini definitions.

astrocfi committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into RF_definitions_190510

juzen2003 committed:
Clean up codes

rfrenchseti committed:
Merge pull request #684 from rfrenchseti/RF_definitions_190510

Update tooltips for various things

juzen2003 committed:
Put back the correct math in updateSliderHandle

dstopp committed:
moved the metadata modal on start to be centered

juzen2003 committed:
Update updateSliderHandle to delete some obs to meet the new row boundary when resizing

juzen2003 committed:
Clean up codes

juzen2003 committed:
Remove unwanted comments

juzen2003 committed:
Remove dataTable, and use data for browse view when we are in table view

juzen2003 committed:
Update to only execute resize logic in updateSliderHandler when it's triggered from window resize

juzen2003 committed:
Fix code style

juzen2003 committed:
Update adjustBrowseHeight

juzen2003 committed:
Update calculateGalleryHeight to fix the gallery and table view height in the cart

juzen2003 committed:
Clean up codes

juzen2003 committed:
Remove spinner in cart tab for infiniteScroll if all data are rendered

juzen2003 committed:
Clean up code style

juzen2003 committed:
Update adjustProductInfoHeight

rfrenchseti committed:
Merge pull request #685 from dstopp/b081_move_metadata_modal

moved the metadata modal on start to be centered

rfrenchseti committed:
Merge branch 'new_ui_infiniteScroll_issues' of https://github.com/juzen2003/pds-opus into juzen2003-new_ui_infiniteScroll_issues

rfrenchseti committed:
Increase browse cache size to 1000 temporarily.

rfrenchseti committed:
Fix non-ASCII character in help string

juzen2003 committed:
- In browse.js, change function parameter "resize" to "browserResized" in
adjustBrowseHeight and updateSliderHandle.
- In browse.js, remove unecessary comments in calculateGalleryHeight.
- In browse.js, fix the updateSliderHandle indentation.

juzen2003 committed:
- In browse.js, in updateSliderHandle function line 707, change "resize"
  to "browserResized".

juzen2003 committed:
- In browse.js, rename setScrollbarOnSlide function to setScrollbarPosition.

juzen2003 committed:
- In browse.js, update event handler function for "request.infiniteScroll"
  in initInfiniteScroll to properly remove spinner (line 1503-1512).

rfrenchseti committed:
Make Galileo orbit numbers have the flyby moon prefix character.

rfrenchseti committed:
Back out centering of metadata modal

rfrenchseti committed:
Merge branch 'juzen2003-new_ui_infiniteScroll_issues' into RF_JS_cleanup_190515

juzen2003 committed:
- In browse.js, update updateBrowseNav to update a global fading flag, it
  is used to determine if the next updateBrowseNav should be called in event
  handler function when clicking ".op-browse-view" (line 101).
- In browse.js, update getBrowseView to take in a "tab" parameter so that
  we know which tab we are at when calling this function.
- In browse.js, update getScrollContainerClass to take in a "tab" parameter
  so that we know which tab we are at when calling this function.
- In browse.js, in loadData, pass in view when calling getStartObsLabel and
  getScrollContainerClass.
- In browse.js, in renderGalleryAndTable, pass in view when calling
  getScrollContainerClass.

rfrenchseti committed:
Merge pull request #687 from rfrenchseti/RF_galileo_orbit_number

Make Galileo orbit numbers have the flyby moon prefix character.

rfrenchseti committed:
Keep various Other target capitalized.

rfrenchseti committed:
Merge pull request #688 from rfrenchseti/RF_capitalize_targets

Keep various Other target capitalized.

juzen2003 committed:
- In browse.js, in updateSliderHandle, update the slider max value to be
  the last page's startObs. It will be used as the max value for slider.

rfrenchseti committed:
Merge branch 'new_ui_infiniteScroll_issues' of https://github.com/juzen2003/pds-opus into juzen2003-new_ui_infiniteScroll_issues

rfrenchseti committed:
Fixes bugs with view arguments

rfrenchseti committed:
Fix one more case of missing view parameter.

rfrenchseti committed:
Merge branch 'juzen2003-new_ui_infiniteScroll_issues' into new_ui

rfrenchseti committed:
Merge branch 'new_ui' into RF_JS_cleanup_190515
